web/source/osk/languageMenu.ts

```diff
@@ -405,22 +409,38 @@ namespace com.keyman.osk {
       kb.innerHTML=unique?kbd['KL']:kbd['KN'].replace(' Keyboard',''); // Name
 
       // We're setting up a few events - this alias helps avoid scoping issues.
-      let languageMenu = this;
+      const languageMenu = this;
 
       // Touchstart (or mspointerdown) event highlights the touched list item
-      var touchStart=function(e) {
+      const touchStart=function(e) {
         e.stopPropagation();
         if(this.className.indexOf('selected') <= 0) {
           this.className=this.className+' selected';
         }
         languageMenu.scrolling=false;
         languageMenu.y0=e.touches[0].pageY;//osk.lgList.childNodes[0].scrollTop;
+
+        // Preserve the original style for the body element; we're going to change
+        // it to block page scrolling.  Must use a separate instance.
+        //
+        // Reference: https://stackoverflow.com/a/28411556
+        languageMenu.originalBodyStyle = {};
+
+        // Must be separate line from previous due to TS type inference stuff.
+        const obs = languageMenu.originalBodyStyle;
+        const dbs = document.body.style;
+        obs.overflowY = dbs.overflowY;
+        obs.height = dbs.height;
+
+        // Now that the properties we're going to overwrite have been cached...
+        dbs.overflowY = 'hidden';
+        dbs.height = '100%';
```

Comment:
Can we factor this into a separate function alongside `restoreBodyStyle()`?

Perhaps we can call them `stopBodyScroll()` and `restoreBodyScroll()`. And please also add a link to the issue in the comment for the functions.